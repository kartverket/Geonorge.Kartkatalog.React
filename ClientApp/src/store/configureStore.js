import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'connected-react-router';
import rootReducer from '../reducers'

//export const history = createBrowserHistory()

//export default function configureStore(initialStore) {

// In development, use the browser's Redux dev tools extension if installed
// if (isDevelopment && typeof window !== 'undefined' && window.devToolsExtension) {
//   enhancers.push(window.devToolsExtension());
//  enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ? window.__REDUX_DEVTOOLS_EXTENSION__ : compose);
// }
/*if (isDevelopment && typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
  enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__());
}*/


//routerMiddleware(history), // for dispatching history actions

const initialState = {};

const middleware = [thunk];

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  rootReducer,  // root reducer with router state
  initialState,
  compose(
    applyMiddleware(...middleware),
    thunk,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
)

export default store;

//const history = createBrowserHistory();

//export history;

