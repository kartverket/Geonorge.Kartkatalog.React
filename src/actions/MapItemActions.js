import { FETCH_MAPITEMS } from './types';

export const fetchMapItems = () => dispatch => {
	let mapItems = localStorage.mapItems && Array.isArray(JSON.parse(localStorage.mapItems)) ? JSON.parse(localStorage.mapItems) : [];
	dispatch({
		type: FETCH_MAPITEMS,
		payload: mapItems
	})
}
