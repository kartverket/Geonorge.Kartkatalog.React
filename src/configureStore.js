import { createBrowserHistory } from 'history'
import { routerMiddleware } from 'connected-react-router'
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import createRootReducer  from './reducers';

export const history = createBrowserHistory()



export default function configureStore(preloadedState) {
	const middleware = [thunk];
	const history = {};
	const store = createStore(
		createRootReducer(history), 
		preloadedState,
		compose (
			applyMiddleware(...middleware, routerMiddleware(history)),
			window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
			) 
		);

	return store;
}

