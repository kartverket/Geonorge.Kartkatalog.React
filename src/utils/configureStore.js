import { createBrowserHistory } from 'history'
import { routerMiddleware } from 'connected-react-router'
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import createRootReducer from '../reducers';

export const history = createBrowserHistory()

const composeEnhancers = composeWithDevTools({
	// options like actionSanitizer, stateSanitizer
});

export default function configureStore(preloadedState) {
	const middleware = [thunk];
	const history = {};
	const store = createStore(
		createRootReducer(history),
		preloadedState,
		composeEnhancers(
			applyMiddleware(...middleware, routerMiddleware(history)),
		)
	);

	return store;
}

