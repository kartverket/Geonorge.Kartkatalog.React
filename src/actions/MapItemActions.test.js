import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

// Actions to be tested
import * as mapItemActions from './MapItemActions';

const middlewares = [thunk] // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);
const store = mockStore();

const mapItem = {
    Title: "title mapItem",
    Uuid: "123456",
}

describe('map_item_actions', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
    });

    describe('fetchMapItems', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(mapItemActions.fetchMapItems());
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    describe('addMapItems', () => {
        test('Add map item', () => {
            store.dispatch(mapItemActions.addMapItem(mapItem)); 
            expect(localStorage.mapItems).toContain("123456");
        });
    });

    describe('removeMapItems', () => {
        test('Remove map item', () => {
            store.dispatch(mapItemActions.removeMapItem(mapItem)); 
            expect(localStorage.mapItems).not.toContain("123456");
        });
    });

});