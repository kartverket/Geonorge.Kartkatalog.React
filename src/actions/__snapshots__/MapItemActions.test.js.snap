// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map_item_actions addMapItems Dispatches the correct action and payload 1`] = `[Function]`;

exports[`map_item_actions fetchMapItems Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Array [],
    "type": "FETCH_MAPITEMS",
  },
]
`;
